contracts/extensions/contracts/BalanceThresholdFilter/MixinBalanceThresholdFilterCore.sol

```diff
@@ -0,0 +1,323 @@
+            case 0x4d0ae54600000000000000000000000000000000000000000000000000000000 { recordAddressesForBatchFillVariant() }    // batchFillOrKillOrders
+            case 0xb4be83d500000000000000000000000000000000000000000000000000000000 { recordAddressesForFillOrderVariant() }    // fillOrder
+            case 0x3e228bae00000000000000000000000000000000000000000000000000000000 { recordAddressesForFillOrderVariant() }    // fillOrderNoThrow
+            case 0x64a3bc1500000000000000000000000000000000000000000000000000000000 { recordAddressesForFillOrderVariant() }    // fillOrKillOrder
+            case 0xe5fa431b00000000000000000000000000000000000000000000000000000000 { recordAddressesForMarketFillVariant() }   // marketBuyOrders
+            case 0xa3e2038000000000000000000000000000000000000000000000000000000000 { recordAddressesForMarketFillVariant() }   // marketBuyOrdersNoThrow
+            case 0x7e1d980800000000000000000000000000000000000000000000000000000000 { recordAddressesForMarketFillVariant() }   // marketSellOrders
+            case 0xdd1c7d1800000000000000000000000000000000000000000000000000000000 { recordAddressesForMarketFillVariant() }   // marketSellOrdersNoThrow
+            case 0x3c28d86100000000000000000000000000000000000000000000000000000000 { recordAddressesForMatchOrders() }         // matchOrders
+            case 0xd46b02c300000000000000000000000000000000000000000000000000000000 {}                                          // cancelOrder
+            case 0x4ac1478200000000000000000000000000000000000000000000000000000000 {}                                          // batchCancelOrders
+            case 0x4f9559b100000000000000000000000000000000000000000000000000000000 {}                                          // cancelOrdersUpTo
+            default {
+                // Revert with `Error("INVALID_OR_BLOCKED_EXCHANGE_SELECTOR")`
+                mstore(0x00, 0x08c379a000000000000000000000000000000000000000000000000000000000)
+                mstore(0x20, 0x0000002000000000000000000000000000000000000000000000000000000000)
+                mstore(0x40, 0x00000024494e56414c49445f4f525f424c4f434b45445f45584348414e47455f)
+                mstore(0x60, 0x53454c4543544f52000000000000000000000000000000000000000000000000)
+                mstore(0x80, 0x00000000)
+                // Revert length calculation:
+                // 4   -- error selector
+                // 32  -- offset to string
+                // 32  -- string length field
+                // 64  -- strlen(INVALID_OR_BLOCKED_EXCHANGE_SELECTOR) rounded up to nearest 32-byte word.
+                revert(0, 132)
+            }
+            // solhint-enable no-empty-blocks
+
+            ///// Validate Recorded Addresses /////
+
+            // Load from memory the addresses to validate
+            let addressesToValidate := mload(0x40)
```

Comment:
It feels safer and more descriptive to make this `let addressesToValidate := mload(add(validatedAddresses, 0x20))` (only really matters if we're manually incrementing the free memory ptr as suggested above).